sdk/compiler/damlc/tests/src/DA/Test/DamlcIntegration.hs

```diff
@@ -391,6 +392,14 @@ data DamlOutput = DamlOutput
   , scriptResults :: HashMap.HashMap T.Text T.Text
   }
 
+stripPartySuffix :: T.Text -> T.Text
+stripPartySuffix = replace "'([a-zA-Z0-9]+)-[a-z0-9]{8}'" "'\\1'"
```

Comment:
The `{8}` here matches the suffix length passed here
https://github.com/digital-asset/daml/pull/20239/files#diff-f5ca808f99c875815abd96d4a6725f970902ebb1ed9668fac396605d922b3841R381

Perhaps this regex could use `+` instead of a specific count, but I was trying to avoid the replacement being used in parts of the input that were not parties.

Reply:
Do those party names only ever come from our tests and thus will never contain names such as `noob-20041011`?